### PR TITLE
Refactor and Enhance Awesome Books App

### PR DESCRIPTION
**Pull request by Zafron047**
This Pull Request refactors and enhances the Awesome Books app according to the following rules:

- [ ] Reimplemented the app using ES6 syntax for improved code readability and maintainability.

- [ ] Divided the code into modular components by creating separate files for each module in the modules directory.

- [ ] Imported the modules into the index.js file to improve code organization and separation of concerns.

- [ ] Ensured all functionalities of the app are maintained without introducing any errors during the refactoring process.

- [ ] Refactored functions to use arrow function syntax for a more concise and modern coding style.

- [ ] Utilized let and const keywords appropriately for variable declaration based on their mutability.

- [ ] Integrated the Luxon library for working with dates and times.

- [ ] Downloaded the Luxon library using npm package manager and imported it as a module.

- [ ] Implemented the display of the current date below the navigation bar using Luxon's functionalities.

- [ ] Demonstrated effective usage of a new tool (Luxon) with minimal externally curated support, simulating real-world scenarios and preparing for future professional engagements.

This Pull Request aims to improve the codebase of the Awesome Books app, enhance its functionalities, and adhere to modern JavaScript development practices.

### CODE REVIEW

**Comment by Zafron047** [comment]
Gratitude in advance for reviewing and contributing to the pull request. Your feedback and contributions are highly appreciated and will enhance the quality and effectiveness of the Awesome Books app. Thank you for your support!

Best regards,

Nohain Islam Zafron 😄

File: modules/bookshelf.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    if (this.DATA.length > 0) {
+      this.show();
+    }
+  }
+
+  add(author, title) {
+    this.DATA = this.DATA.concat({ author, title });
+    localStorage.setItem('data', JSON.stringify(this.DATA));
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+  remove(author, title) {
+    this.DATA = this.DATA.filter(
+      (b) => b.author !== author || b.title !== title,
+    );
+    localStorage.setItem('data', JSON.stringify(this.DATA));
+    this.counter -= 1;
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      }
+
+      this.section.appendChild(div);
+
+      removeBtn.addEventListener('click', () => {
+        this.remove(data.author, data.title);
+        this.show();
+      });
+    });
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+  init() {
+    this.form.addEventListener('submit', (e) => {
+      e.preventDefault();
+      this.add(this.authorInput.value, this.titleInput.value);
+      this.form.reset();
+      this.show();
+    });
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+  showListPage() {
+    this.list.addEventListener('click', () => {
+      this.booksWrapper.style.display = 'block';
+      this.form.style.display = 'none';
+      this.contactPage.style.display = 'none';
+    });
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+  showFormPage() {
+    this.addNew.addEventListener('click', () => {
+      this.form.style.display = 'flex';
+      this.booksWrapper.style.display = 'none';
+      this.contactPage.style.display = 'none';
+    });
+  }
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

File: modules/bookshelf.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+  showContactPage() {
+    this.contact.addEventListener('click', () => {
+      this.contactPage.style.display = 'flex';
+      this.form.style.display = 'none';
+      this.booksWrapper.style.display = 'none';
+    });
+  }
+}
```
Comment (binhussen) [inline modules/bookshelf.js]: - Great job for implementing using class, I think in this milestone you have to use an arrow function to fit the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md) 👍 
Screenshot 2023-06-05 154539

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Zafron047 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Well Stractured Code✔️
- Sticked Given WIreframe ✔️
- Used GitHub flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
Hello @binhussen, 

Thank you for pointing out the issue. I have made the necessary changes to the code by converting all the regular functions inside the class to Arrow functions. This should address the concern you raised.

Best regards,
[Your Name]

**Review by wayungi — APPROVED** [review]
Hi @Zafron047 , 

image

Great work, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Remember: Every finish line 🏁  is the beginning of a new race 🐎** 

## Highlights
- [ ] Arrow functions used ✔️ 
- [ ]  Correct GitHub flow ✔️ 
- [ ] Descriptive PR body ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Zafron047** [comment]
@wayungi Thanks a lot for giving your time to review it.
Have a great day. 👍 😄

**Pull request merged**